src/shared/Microsoft.Git.CredentialManager/BrowserUtils.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Git.CredentialManager
+{
+    public static class BrowserUtils
+    {
+        public static void OpenDefaultBrowser(IEnvironment environment, string url)
+        {
+            if (!Uri.TryCreate(url, UriKind.Absolute, out Uri uri))
+            {
+                throw new ArgumentException($"Not a valid URI: '{url}'");
+            }
+
+            OpenDefaultBrowser(environment, uri);
+        }
+
+        public static void OpenDefaultBrowser(IEnvironment environment, Uri uri)
+        {
+            if (!uri.Scheme.Equals(Uri.UriSchemeHttp, StringComparison.OrdinalIgnoreCase) &&
+                !uri.Scheme.Equals(Uri.UriSchemeHttps, StringComparison.OrdinalIgnoreCase))
```

Comment:
Should this be OR rather than AND?

Reply:
We're doing a `!` not on each `Equals` call, so this is checking:

```
Scheme != HTTP && Scheme != HTTPS
```

..i.e., the scheme is something other than HTTP(S).

[De Morgan-ising](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) it would be:

```csharp
if ( ! ( scheme == HTTP || scheme == HTTPS ) )
{
    throw new ArgumentException(..);
}
```

Reply:
Got it!